odoo/fields.py

```diff
@@ -2267,27 +2278,32 @@ def process(ids):
         elif isinstance(value, (list, tuple)):
             # value is a list/tuple of commands, dicts or record ids
             comodel = record.env[self.comodel_name]
+            if record and not record.id:
+                # record is new: the value of self is new records
+                browse = lambda id_: comodel.new(origin=comodel.browse(id_))
+            else:
+                browse = comodel.browse
             # determine the value ids; by convention empty on new records
             ids = OrderedSet(record[self.name].ids if record.id else ())
```

Comment:
Could you check for this line the proposal I made in https://github.com/odoo/odoo/pull/24458? thanks